### ISSUE TITLE
Should we run some basic checks after parsing a model?

### ISSUE THREAD

**Issue by MichaelClerx** [ModellingWebLab/cellmlmanip#242]
At the moment we have some methods to check the model makes sense:

- `Model.check_left_right_units_equal`

and 

- `testparser.py::check_rdf_identities(model)`
- `testparser.py::check_dummy_assignment(model)`

(I moved these last two into testing at one stage, as that was the only place they were used)

Should these be standard checks that occur after parsing a model? And maybe all hidden but called by some `validate()` method?

**Comment by MichaelClerx** [comment]
Thoughts @jonc125 @MauriceHendrix @skeating ?

**Comment by MauriceHendrix** [comment]
It's not a bad idea but I do want to be a bit careful.

Depending on the validation my tests might be harder (probably not but I'd want to test it first)
Also: will there be a performance implication?
Is there a chance of disallowing something that's perfectly legal (like unrelated metadata etc)?

In any case I'd prefer to have an option to bypass the validation if I felt like it.

**Comment by MichaelClerx** [comment]
I'm guessing there won't be much of a performance hit

But we could definitely make the validation optional. Have some kind `validate` method that returns True or False.

I'd want that anyway for the use case where users build a model from scratch through the API, or when they modify a model after loading it

**Comment by MauriceHendrix** [comment]
> I'm guessing there won't be much of a performance hit
> 
> But we could definitely make the validation optional. Have some kind `validate` method that returns True or False.
> 
> I'd want that anyway for the use case where users build a model from scratch through the API, or when they modify a model after loading it

I don't might if the deaful is on for the validation

**Comment by jonc125** [comment]
Happy to start curating a list of possible checks, but let's not implement any yet. Some like thorough units checking would fail for most existing models, for instance... so we need to be selective!

**Comment by MauriceHendrix** [comment]
I guess that's what warnings are for?

**Comment by skeating** [comment]
Yes but other people who want things to be strict and fail completely if there are discrepancies. Modellers come in all shapes and sizes 😄

**Comment by MichaelClerx** [comment]
Also if you're using cellmlmanip you wouldnt' want to filter warnings to see which ones to keep as warnings and which ones to show as an error to the user

**Comment by MauriceHendrix** [comment]
> Yes but other people who want things to be strict and fail completely if there are discrepancies. Modellers come in all shapes and sizes 😄

ha ha sure that's fair enough
